src/communication/communicator.hpp

```diff
@@ -45,31 +48,71 @@ class communicator {
 
     communicator() {}
 
-    explicit communicator(gid_partition_type cell_gid_partition):
-        cell_gid_partition_(cell_gid_partition)
-    {}
-
-    cell_local_size_type num_groups_local() const
-    {
-        return cell_gid_partition_.size();
-    }
-
-    void add_connection(connection con) {
-        EXPECTS(is_local_cell(con.destination().gid));
-        connections_.push_back(con);
-    }
-
-    /// returns true if the cell with gid is on the domain of the caller
-    bool is_local_cell(cell_gid_type gid) const {
-        return algorithms::in_interval(gid, cell_gid_partition_.bounds());
-    }
+    explicit communicator(const recipe& rec, const domain_decomposition& dom_dec) {
```

Comment:
So, the choices are: recipe, domain -> communicator, domain -> recipe -> communicator, recipe -> domain -> communicator. Any arguments for this approach?